vendor/bat-native-ledger/src/bat/ledger/internal/gemini/gemini.cc

```diff
@@ -0,0 +1,289 @@
+    ContributionCompleted(type::Result::LEDGER_ERROR, "", contribution_id,
+                          amount, "", callback);
+    return;
+  }
+
+  const double fee = (amount * 1.05) - amount;
+  const double reconcile_amount = amount - fee;
+
+  auto contribution_callback =
+      std::bind(&Gemini::ContributionCompleted, this, _1, _2, contribution_id,
+                fee, info->publisher_key, callback);
+
+  Transaction transaction;
+  transaction.address = info->address;
+  transaction.amount = reconcile_amount;
+
+  transfer_->Start(transaction, contribution_callback);
+}
+
+void Gemini::ContributionCompleted(const type::Result result,
+                                   const std::string& transaction_id,
+                                   const std::string& contribution_id,
+                                   const double fee,
+                                   const std::string& publisher_key,
+                                   ledger::ResultCallback callback) {
+  if (result == type::Result::LEDGER_OK) {
+    SaveTransferFee(contribution_id, fee);
+
+    if (!publisher_key.empty()) {
+      ledger_->database()->UpdateContributionInfoContributedAmount(
+          contribution_id, publisher_key, callback);
+      return;
+    }
+  }
+
+  callback(result);
+}
+
+void Gemini::FetchBalance(FetchBalanceCallback callback) {
+  const auto wallet = GetWallet();
+
+  if (!wallet || wallet->token.empty() || wallet->address.empty()) {
+    callback(type::Result::LEDGER_OK, 0.0);
+    return;
+  }
+
+  if (wallet->status == type::WalletStatus::CONNECTED) {
+    BLOG(1, "Wallet is connected");
+    callback(type::Result::LEDGER_OK, 0.0);
+    return;
+  }
+
+  auto url_callback =
+      std::bind(&Gemini::OnFetchBalance, this, _1, _2, callback);
+
+  gemini_server_->post_balance()->Request(wallet->token, url_callback);
+}
+
+void Gemini::OnFetchBalance(const type::Result result,
+                            const double available,
+                            FetchBalanceCallback callback) {
+  if (result == type::Result::EXPIRED_TOKEN) {
+    BLOG(0, "Expired token");
+    DisconnectWallet();
+    callback(type::Result::EXPIRED_TOKEN, 0.0);
+    return;
+  }
+
+  if (result != type::Result::LEDGER_OK) {
+    BLOG(0, "Couldn't get balance");
+    callback(type::Result::LEDGER_ERROR, 0.0);
+    return;
+  }
+
+  callback(type::Result::LEDGER_OK, available);
+}
+
+void Gemini::TransferFunds(const double amount,
+                           const std::string& address,
+                           client::TransactionCallback callback) {
+  Transaction transaction;
+  transaction.address = address;
+  transaction.amount = amount;
+  transfer_->Start(transaction, callback);
+}
+
+void Gemini::WalletAuthorization(
+    const base::flat_map<std::string, std::string>& args,
+    ledger::ExternalWalletAuthorizationCallback callback) {
+  authorization_->Authorize(args, callback);
+}
+
+void Gemini::GenerateWallet(ledger::ResultCallback callback) {
```

Comment:
`ledger` namespace not required here, maybe other places too

Reply:
fixed - https://github.com/brave/brave-core/pull/8566/files#diff-4bd796bc3bb28fd3579feb8625aa26eb4100fe6c46162c88ec36ec54f3b74f00R156